### ISSUE TITLE
Generate icons in parallel using `rayon` and `num_cpus`

### ISSUE THREAD

**Issue by ngirard** [GarkGarcia/icon-pie#13]
It would be nice if the icons would be generated in parallel. This can be achieved with very little cost using the `rayon` and `num_cpus` crates.

There are useful examples [here](https://rust-lang-nursery.github.io/rust-cookbook/concurrency.html).

**Comment by GarkGarcia** [comment]
I easy to do, but I don't really know if it would benefit the app. The app already runs pretty smoothly and I believe [`image`](https://crates.io/crates/image) (which [`icon_baker`](https://crates.io/crates/icon_baker) uses for raster graphics) is gpu accelerated.

I'll have to think a bit more about it.

**Comment by ngirard** [comment]
I wouldn't tell myself for sure, but it seems like it might be beneficial for large icon themes generated from SVG files.

KDE's icon themes count [thousands of files](https://github.com/RazrFalcon/resvg#performance) (3417 files for the Elementary icon theme ; 4947 files for the Oxygen icon theme) ; and the Breeze icon theme for LibreOffice counts more than 2500 files.

**Comment by GarkGarcia** [comment]
Makes sense. I'll work on it when I get some spare time. Help would be appreciated.

**Comment by GarkGarcia** [comment]
This seems like something that should be implemented in `icon_baker`. I believe [this](https://docs.rs/icon_baker/3.0.1/src/icon_baker/lib.rs.html#151) is the relevant method that should be altered.

**Comment by GarkGarcia** [comment]
This is our biggest priority right now. Images should definitively be resampled in parallel. Our main issue is that `usvg::Tree` and `image::DynamicImage` don't implement `Send`, so it's hard to pass them around to different threads.
